src/output.cpp

```diff
@@ -1383,8 +1458,8 @@ void draw_subtab(WINDOW *w, int iOffsetX, std::string sText, bool bSelected, boo
 }
 
 void draw_scrollbar(WINDOW *window, const int iCurrentLine, const int iContentHeight,
-                    const int iNumEntries, const int iOffsetY, const int iOffsetX,
-                    nc_color bar_color)
+                    int iNumEntries, const int iOffsetY, const int iOffsetX,
+                    nc_color bar_color, bool bRefresh, int iFakeEntries)
```

Comment:
> If there are 10 items/lines in a list but only 5 rows to display them, you can scroll 10 times with the current behaviour, which is ok because you want to be able to select every single line.

> For displaying only text, without selecting anything, I only want to scroll 5 times. Otherwise you would get blank lines at the bottom like you said. But as it was, if numentries (5) is less or equal to display height(5) it won't let me scroll. Couldn't come up with a better solution.

Ah, I think the problem is different meaning here: scrolling != selecting. Those are two different things. Scrolling is for the printed lines (after folding and all), selecting is for items of a list.

The data (to be displayed) may contain items that spawn multiple lines when printed (or an variable amount of lines, like the message entries in the message log) or lines that don't belong to any item (the category headers in the AIM view in category sort). Those lines can not be selected, but they could be scrolled through.

The scrollbar should represent the scrolling of the visible area, it has nothing to do with the selected item itself. Selected item and scroll-offset are different things (GUIs usually allow scrolling without changing the selection). I fear, this separation is not always done correctly in the existing code.

`draw_scrollbar` needs to know:
- where it should be drawn (`iContentHeight`, offsets, window, color),
- the current scroll-offset (`iCurrentLine`),
- the minimal scroll-offset (not a parameter, implicitly `0`),
- the maximal scroll-offset (`iNumEntries`-1).

You have a text of `N` lines, printed in a window containing `H` writable lines (excluding the border). The scroll-offset is in the range `[0...std::max(0, N-H)]`. `iNumEntries` would be `std::max(0, N-H)+1`. (`max` to catch the case when scrolling is not needed at all because `N <= H`)

It seems you have to call `draw_scrollbar` like this:
```C++
draw_scrollbar(win, selected, height, std::max(0, iLines - height) + 1, 1, 0, c_white);
```